src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothLeAdapterImpl.cs

```diff
@@ -346,40 +392,58 @@ internal int FreeServiceDataList()
 
         internal int GetScanResultAppearance(BluetoothLeScanData scanData, BluetoothLePacketType packetType)
         {
-            int appearance;
+            int appearance = -1;
 
-            BluetoothLeScanDataStruct scanDataStruct = BluetoothUtils.ConvertLeScanDataToStruct (scanData);
+            if (BluetoothAdapter.IsBluetoothEnabled && Globals.IsInitialize)
+            {
+                BluetoothLeScanDataStruct scanDataStruct = BluetoothUtils.ConvertLeScanDataToStruct(scanData);
 
-            int ret = Interop.Bluetooth.GetScanResultAppearance(ref scanDataStruct, packetType, out appearance);
-            if (ret != (int)BluetoothError.None) {
-                Log.Error(Globals.LogTag, "Failed to get Appearance value- " + (BluetoothError)ret);
+                int ret = Interop.Bluetooth.GetScanResultAppearance(ref scanDataStruct, packetType, out appearance);
+                if (ret != (int)BluetoothError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get Appearance value- " + (BluetoothError)ret);
+                    BluetoothErrorFactory.ThrowBluetoothException(ret);
+                }
             }
+            else
+            {
+                BluetoothErrorFactory.ThrowBluetoothException((int)BluetoothError.NotEnabled);
+            }
+
             return appearance;
         }
 
         internal ManufacturerData GetScanResultManufacturerData(BluetoothLeScanData scanData, BluetoothLePacketType packetType)
         {
-            int dataId = 0;
-            int dataLength = 0;
-            IntPtr manufData;
-
-            BluetoothLeScanDataStruct scanDataStruct = BluetoothUtils.ConvertLeScanDataToStruct (scanData);
             ManufacturerData data = new ManufacturerData();
 
-            int ret = Interop.Bluetooth.GetScanResultManufacturerData(ref scanDataStruct, packetType, out dataId,
-                out manufData, out dataLength);
-            if (ret != (int)BluetoothError.None)
+            if (BluetoothAdapter.IsBluetoothEnabled && Globals.IsInitialize)
             {
-                Log.Error(Globals.LogTag, "Failed to get Manufacturer data - " + (BluetoothError)ret);
-                return null;
-            }
+                int dataId = 0;
+                int dataLength = 0;
+                IntPtr manufData;
+
+                BluetoothLeScanDataStruct scanDataStruct = BluetoothUtils.ConvertLeScanDataToStruct(scanData);
+
+                int ret = Interop.Bluetooth.GetScanResultManufacturerData(ref scanDataStruct, packetType, out dataId, out manufData, out dataLength);
```

Comment:
Accoding native API reference (https://developer.tizen.org/dev-guide/tizen-iot-headless/latest/group__CAPI__NETWORK__BLUETOOTH__ADAPTER__LE__MODULE.html#gaa63e042a226aab567a982bbbdb8240df)

> The manufacturer_data must be released with free() by you.

But, you not

Reply:
Done.